iotdb-core/tsfile/src/main/java/org/apache/iotdb/tsfile/read/filter/GroupByMonthFilter.java

```diff
@@ -38,6 +38,12 @@ public class GroupByMonthFilter extends GroupByFilter {
 
   private int slidingStepsInMo;
   private int intervalInMo;
+
+  private long fixedIntervalOther;
+  private long fixedSlidingStepOther;
+  private long[] timeArray;
+  private int arraySize;
```

Comment:
we should judge it dynamicly.

Reply:
I am not quite sure what this mean, the **timeArray** is an array that stores the start time of each window interva. In this case, I can get the return value of getTimePointPosition directly from **timeArray** when the input is '1mo1d', instead of iteratively calculating from initialStartTime every time